File: python/src/0217-containsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        dict = {}
        numsLen = len(nums)
        for i in range(0, numsLen):
            item = nums[i]
            if (item in dict):
                return True
            else:
                dict[item] = True
        return False

File: typescript/src/0217-containsDuplicate.ts
function containsDuplicate(nums: number[]): boolean {
  const map: object = {};
  const len: number = nums.length;
  for (let i = 0; i < len; i++) {
    const item: number = nums[i];
    if (!map[item]) {
      map[item] = true;
    } else {
      return true;
    }
  }
  return false;
}

export {containsDuplicate};
